### ISSUE TITLE
Blueprints for sub-classing

### ISSUE THREAD

**Issue by trentmwillis** [ahmadsoe/ember-highcharts#11]
I think it would be beneficial if we added blueprints to help with sub-classing the default high-charts component (e.g., if I wanted to create a re-usable pie-chart component). It could be something like:

    ember generate chart pie-chart

Thoughts? If it seems like a good idea, I'll create a PR for the feature.

**Comment by ahmadsoe** [comment]
@trentmwillis sounds good, but do we really need this? as if I want to create pie-chart I just need to specify it in the chart options. also, we don't want to maintenance all of those chart type.

**Comment by trentmwillis** [comment]
Sorry, my example was unclear. Above, what I really meant was:

    ember generate chart <chart-name>

So you could do: `ember generate chart my-cool-chart` or `ember generate chart complex-chart`. Basically, allow the users an easy way to define sub-classes of the Highchart component. We wouldn't need to maintain different types, just the scaffolding for the blueprint.

As a reference: [Ember-CLI Generators & Blueprints](http://www.ember-cli.com/#generators-and-blueprints)

**Comment by ahmadsoe** [comment]
Maybe good, but I don't see many ember addon using this.
What do you think @Dhaulagiri @lolmaus ?

**Comment by lolmaus** [comment]
If it was my project, i would happily accept a PR, but wouldn't bother implementing this feature myself. It saves user literally one line of code:

```
ember g component my-chart
```

```js
import HighCharts from 'ember-highcharts/addon/components/high-charts';

export default HighCharts.extend({
  /* customization here */
});
```

**Comment by ahmadsoe** [comment]
Yeah, i would happily accept a PR too..

@trentmwillis looking forward your PR for this feature then..

**Comment by ahmadsoe** [comment]
Closed in #21
